src/components/quizz/Quizz.js

```diff
@@ -1,12 +1,134 @@
-import React from 'react';
+import React from "react";
+import { quizData } from "../../data/questions";
 import './Quizz.css';
 
-function Quizz () {
-  return (
-    <div>
-      <p>Quizz page</p>
-    </div>
-  );
+class MainQuiz extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      currentQuestion: 0,
+      myAnswer: null,
+      options: [],
+      score: 0,
+      disabled: true,
+      isEnd: false
+    };
+  }
+
+  loadQuizData = () => {
+    this.setState(() => {
+      return {
+        questions: quizData[this.state.currentQuestion].question,
+        answer: quizData[this.state.currentQuestion].answer,
+        options: quizData[this.state.currentQuestion].options
+      };
+    });
+  };
+
+  componentDidMount () {
+    this.loadQuizData();
+  }
+
+  nextQuestionHandler = () => {
+    const { myAnswer, answer, score } = this.state;
+
+    if (myAnswer === answer) {
+      this.setState({
+        score: score + 1
+      });
+    }
+
+    this.setState({
+      currentQuestion: this.state.currentQuestion + 1
+    });
+  };
+
+  componentDidUpdate (prevProps, prevState) {
+    if (this.state.currentQuestion !== prevState.currentQuestion) {
+      this.setState(() => {
+        return {
+          disabled: true,
+          questions: quizData[this.state.currentQuestion].question,
+          options: quizData[this.state.currentQuestion].options,
+          answer: quizData[this.state.currentQuestion].answer
+        };
+      });
+    }
+  }
+  //check answer
+  checkAnswer = answer => {
+    this.setState({ myAnswer: answer, disabled: false });
+  };
+
+  finishHandler = () => {
+    if (this.state.currentQuestion === quizData.length - 1) {
+      this.setState({
+        isEnd: true
```

Comment:
pourquoi avoir un state `isEnd` si on peut déjà avoir l'info avec `this.state.currentQuestion === quizData.length - 1` ?